src/DependencyInjection/DoctrineExtension.php

```diff
@@ -300,8 +300,15 @@ protected function loadDbalConnection($name, array $connection, ContainerBuilder
             $def->setClass($options['wrapperClass']);
         }
 
-        if (! empty($connection['use_savepoints'])) {
-            $def->addMethodCall('setNestTransactionsWithSavepoints', [$connection['use_savepoints']]);
+        if (isset($connection['use_savepoints'])) {
+            // DBAL >= 4 always has savepoints enabled. So we only need to call "setNestTransactionsWithSavepoints" for DBAL < 4
+            if (method_exists(Connection::class, 'getEventManager')) {
+                if ($connection['use_savepoints']) {
+                    $def->addMethodCall('setNestTransactionsWithSavepoints', [$connection['use_savepoints']]);
+                }
+            } elseif (! $connection['use_savepoints']) {
+                throw new LogicException('The "use_savepoints" option can only be set to "true" when using DBAL >= 4');
```

Comment:
Note that this is technically a breaking change for people on DBAL 4 using `use_savepoints = false`. As `setNestTransactionsWithSavepoints` was never called in those cases it did not result in any error. But this seems odd to me so I think having an error in those cases is the correct behavior.